File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.py
class Solution:
    def minimumRounds(self, tasks: List[int]) -> int:
        dict, ans = {}, 0
        for task in tasks:
            if task not in dict:
                dict[task] = 0
            dict[task] += 1
        
        for key,val in dict.items():
            if val == 1:
                return -1
            ans += (val//3) if val % 3 == 0 else ((val//3) + 1 )
        return ans
        

File: 2244-minimum-rounds-to-complete-all-tasks/2244-minimum-rounds-to-complete-all-tasks.ts
function minimumRounds(tasks: number[]): number {
    const map:Record<number, number> = {}
    let ans = 0;
    for(const task of tasks){
        map[task] > 0 ? map[task]++ : map[task] = 1
    }
    for( const obj in map){
        if(map[obj] < 2) return -1;
        ans += Math.ceil(map[obj] / 3);
    }
    return ans
};